File: front/src/services/about/AboutService.ts
import type { AboutPage, AboutPageBlockAlign } from './types.ts'

export interface AboutService {
  aboutPage(): Promise<AboutPage>
  updateBody(body: string): Promise<AboutPage>
  addCanvas(): Promise<AboutPage>
  updateCanvasHeight(id: string, height: number): Promise<AboutPage>
  updateCanvasBackground(id: string, mimeType: string, imageBase64: string): Promise<AboutPage>
  removeCanvasBackground(id: string): Promise<AboutPage>
  removeCanvas(id: string): Promise<AboutPage>
  addTextBlock(
    canvasId: string,
    text: string,
    x: number,
    y: number,
    width: number,
    height: number,
  ): Promise<AboutPage>
  addPhotoBlock(
    canvasId: string,
    mimeType: string,
    imageBase64: string,
    x: number,
    y: number,
    width: number,
    height: number,
  ): Promise<AboutPage>
  addButtonBlock(
    canvasId: string,
    text: string,
    linkUrl: string,
    x: number,
    y: number,
    width: number,
    height: number,
  ): Promise<AboutPage>
  updateBlockLayout(id: string, x: number, y: number, width: number, height: number): Promise<AboutPage>
  updateBlockAlign(id: string, align: AboutPageBlockAlign): Promise<AboutPage>
  updateBlockText(id: string, text: string): Promise<AboutPage>
  updateBlockLink(id: string, linkUrl: string): Promise<AboutPage>
  removeBlock(id: string): Promise<AboutPage>
}

File: back/src/main/kotlin/com/jvmvstv_v/back/about/service/AboutService.kt
package com.jvmvstv_v.back.about.service

import com.jvmvstv_v.back.about.model.AboutPage
import com.jvmvstv_v.back.about.model.AboutPageBlockAlign
import com.jvmvstv_v.back.about.model.AboutPageLanguage
import java.util.UUID

interface AboutService {
    fun aboutPage(): AboutPage
    fun updateBody(body: String, language: AboutPageLanguage): AboutPage
    fun addCanvas(): AboutPage
    fun updateCanvasHeight(id: UUID, height: Double): AboutPage
    fun updateCanvasBackground(id: UUID, mimeType: String, imageBase64: String): AboutPage
    fun removeCanvasBackground(id: UUID): AboutPage
    fun removeCanvas(id: UUID): AboutPage
    fun addTextBlock(canvasId: UUID, text: String, x: Double, y: Double, width: Double, height: Double): AboutPage
    fun addPhotoBlock(
        canvasId: UUID,
        mimeType: String,
        imageBase64: String,
        x: Double,
        y: Double,
        width: Double,
        height: Double,
    ): AboutPage
    fun addButtonBlock(
        canvasId: UUID,
        text: String,
        linkUrl: String,
        x: Double,
        y: Double,
        width: Double,
        height: Double,
    ): AboutPage
    fun updateBlockLayout(id: UUID, x: Double, y: Double, width: Double, height: Double): AboutPage
    fun updateBlockAlign(id: UUID, align: AboutPageBlockAlign): AboutPage
    fun updateBlockText(id: UUID, text: String, language: AboutPageLanguage): AboutPage
    fun updateBlockLink(id: UUID, linkUrl: String): AboutPage
    fun removeBlock(id: UUID): AboutPage
}
